CONTRIBUTION.md

```diff
@@ -0,0 +1,245 @@
+  repository:
+
+    * **Bug fixes** should be based on the branch named after the **oldest
+      supported release line** the bug affects.
+
+        * E.g. if a feature was introduced in 1.1, the latest release line is
+          1.3, and a bug is found in that feature - make your branch based on
+          1.1.  The maintainer will then forward-port it to 1.3 and master.
+        * Bug fixes requiring large changes to the code or which have a chance
+          of being otherwise disruptive, may need to base off of **master**
+          instead. This is a judgement call -- ask the devs!
+
+    * **New features** should branch off of **the 'master' branch**.
+
+        * Note that depending on how long it takes for the dev team to merge
+          your patch, the copy of ``master`` you worked off of may get out of
+          date! If you find yourself 'bumping' a pull request that's been
+          sidelined for a while, **make sure you rebase or merge to latest
+          master** to ensure a speedier resolution.
+
+Code formatting
+---------------
+
+* **Follow the style you see used in the primary repository**! Consistency with
+  the rest of the project always trumps other considerations. It doesn't matter
+  if you have your own style or if the rest of the code breaks with the greater
+  community - just follow along.
+
+Documentation isn't optional
+----------------------------
+
+It's not! Patches without documentation will be returned to sender. a separate commit should be added that contains the ouput from running `npm run docs`. Methods and classes should also have jsdoc documentation.
+
+example:
+```javascript
+/**
+ * increment number by 1
+ * @param n - number to be incremented
+ */
+increment(n) {
+  return n + 1;
+}
+```
+
+Tests aren't optional
+---------------------
+
+Any bugfix that doesn't include a test proving the existence of the bug being
+fixed, may be suspect.  Ditto for new features that can't prove they actually
+work.
+
+Your ideally include unit tests and/or storybooks to test and help describe what you are solving
+
+We've found that test-first development really helps make features better
+architected and identifies potential edge cases earlier instead of later.
+Writing tests before the implementation is strongly encouraged.
+
+Full example
+------------
+
+Here's an example workflow for a project ``theproject`` hosted on Github, which
+is currently in version 1.3.x. Your username is ``yourname`` and you're
+submitting a basic bugfix.
+
+Preparing your Fork
+^^^^^^^^^^^^^^^^^^^
+
+1. Click 'Fork' on Github, creating e.g. ``yourname/theproject``.
+2. Clone your project: ``git clone git@github.com:yourname/theproject``.
+3. ``cd theproject``
+4. Install the development requirements: ``npm install``.
+5. Create a branch: ``git checkout -b foo-the-bars 1.3``.
+
+Making your Changes
+^^^^^^^^^^^^^^^^^^^
+
+1. Add changelog entry crediting yourself.
```

Comment:
No, I currently use release notes and look over pull requests. It is a decent way to keep track of changes, but I think it would be easier to look at the changelog instead of pr history